tests/e2e-pw/specs/shopper/shopper-checkout-failures.spec.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * External dependencies
+ */
+import { test, expect, Page } from '@playwright/test';
+
+/**
+ * Internal dependencies
+ */
+
+import { config } from '../../config/default';
+import * as shopper from '../../utils/shopper';
+
+test.describe( 'Shopper > Checkout > Failures with various cards', () => {
+	const waitForBanner = async ( page: Page, errorText: string ) => {
+		await expect( page.getByText( errorText ) ).toBeVisible();
+	};
+
+	test.beforeEach( async ( { page } ) => {
+		await shopper.addCartProduct( page );
+
+		await page.goto( '/checkout/' );
+		await shopper.fillBillingAddress(
+			page,
+			config.addresses.customer.billing
+		);
```

Comment:
I suggest replacing the redirect and billing address autofill with shopper.setupCheckout(), which handles those tasks while also performing additional checks.

```suggestion
		await shopper.setupCheckout( page, config.addresses.customer.billing );
```

Reply:
Thanks for this suggestion, I've added it in [0e93bb6](https://github.com/Automattic/woocommerce-payments/pull/10105/commits/0e93bb6dc0279f0b2cd52652bf4f69fafc8d88a7).